docs/courses/csintro3/functions/comments.md

```diff
@@ -1,9 +1,213 @@
+
+## Concept: Commenting Simple Functions
+
+The first functions discussed in the [Intro Lesson](/courses/csintro3/functions/intro) were fairly basic; they accepted no input, and did not produce any output.
+
+The ``||functions:sayHello||`` function is a simple demonstration of this; whenever it is called, it will ``||game:splash||`` "hello" to the person playing the game.
+
+To add a comment to a function like this, simply describe in a short sentence what the function is actually doing.
+
+## Example #1: Commenting on ``||functions:sayHello||``
+
+1. Review the code below
+2. Identify what the method will do
+3. Read the comment for the method: identify how it describes the method's behavior
+
+```typescript
+/**
+ * Displays a greeting for the player to read.
+ */
+function sayHello() {
+    game.splash("hello");
+}
+
+sayHello();
+```
+
+Notice that the comment does not go into much detail on the behavior - saying the exact greeting, or how it is done. If someone wants to know these things, they can read the code itself.
+
+Even if the function were changed to display different text in a slightly different way (as done below), the same comment could still be used to describe the function.
+
+```typescript
 /**
- * DESCRIPTION OF METHOD VALUE
- * param pname: what it do
- * returns: what it returns
- */
\ No newline at end of file
+ * Displays a greeting for the player to read.
+ */
+function sayHello() {
+    game.showLongText("Welcome to our new Adventure! You will have a good time here!", DialogLayout.Bottom);
+}
+
+sayHello();
+```
+
+## Student Task #1: Writing a Comment
+
+1. Review the code below
+2. Fill in the comment for the ``||functions:helloWorld||`` function
+3. Fill in the comment for the ``||functions:goodbye||`` function
+4. **Challenge:** should a description of ``||variables:word||`` show up in the comment for ``||functions:goodbye||``?
+
+```typescript
+/**
+ * 
+ */
+function helloWorld() {
+    game.splash("Hello World!");
+}
+
+/**
+ * 
+ */
+function goodbye() {
+    let word = "Goodbye!";
+    game.splash(word);
+}
+```
+
+### ~hint
+
+Hints for Challenge:
+
+* Is ``||variables:word||`` accessible outside of ``||functions:goodbye||``?
+* How is the behavior of ``||functions:goodbye||`` different from ``||functions:helloWorld||``? Is ``||variables:word||`` part of that difference?
+
+### ~
+
+## Concept: Parameters
+
+To comment on a parameter, start the line with ``@param``, followed by the parameter's name, and then a short description of what the method will do.
+
+For example, a section of a comment describing a parameter ``count`` of type ``number`` that determines how many times a value will be printed could be commented on as
+
+```typescript-ignore
+/**
+ * Omitted function description
```

Comment:
You describe what this function does so maybe use this as an opportunity to show another example of a good comment.

Or drop the word "Omitted". It's kind of confusing. Maybe simpler to just say "Some function description"

Reply:
oops I changed that locally already, good catch